algo-core/scheduler_utils.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import json
+import sys
+from pathlib import Path
+import jsonschema
+import pandas as pd
+
+
+# Input filenames:
+filename_onboarding = "onboarding_agents.txt"
+filename_mentors = "mentors.txt"
+filename_new = "new_agents.txt"
+input_folder = ""
+
+def get_project_root() -> Path:
+    return Path(__file__).parent.parent
+
+
+def parse_json_input():
+    """Read, validate and return json input."""
+    # Production (read input from command line):
+    global input_folder
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i", "--input", help="Scheduler input JSON file path", required=True
+    )
+    args = parser.parse_args()
+    input_filename = args.input.strip()
+
+    # Testing (define input directly):
+    # input_filename = 'support-shift-scheduler-input.json'
+
+    # Load and validate JSON input:
+    input_json = json.load(open(input_filename))
+    path_to_schema = Path(get_project_root() / 'lib/schemas/' , 'support-shift-scheduler-input.schema.json')
```

Comment:
I like the approach but this mixed syntax is somewhat confusing IMO. Besides, appending `'lib/schemas/'` means that we're forcing the unix-style `/` separator instead of letting the `Path` decide based on the system type.

Anyway, since `get_project_root()` returns a `Path` object, we can concatenate pieces together using the `/` *operator*, which is a syntactic sugar to build `Path` objects:
```suggestion
    path_to_schema = get_project_root() / 'lib' / 'schemas' / 'support-shift-scheduler-input.schema.json'
```

Alternatively, we can pass the same pieces to the `Path` constructor to maybe make it more explicit
```suggestion
    path_to_schema = Path(get_project_root(), 'lib', 'schemas' , 'support-shift-scheduler-input.schema.json')
```

Both results are the same so it's a matter of style. However, I'd suggest leaning towards one for the sake of consistency.